src/checkpoint/container/queue_serialize.h

```diff
@@ -51,8 +51,49 @@
 
 namespace checkpoint {
 
+template<typename SerializerT, typename T>
+void deserializeQueueElems(SerializerT& s, std::queue<T>& q, typename std::queue<T>::size_type size) {
+  using Reconstructor =
+    dispatch::Reconstructor<typename dispatch::CleanType<T>::CleanT>;
+
+  dispatch::Allocator<T> allocated;
+  for (typename std::queue<T>::size_type i = 0; i < size; ++i) {
+    auto* reconstructed = Reconstructor::construct(allocated.buf);
+    s | *reconstructed;
+    q.push(std::move(*reconstructed));
+  }
+}
+
+template<typename SerializerT, typename T>
+void serializeQueueElems(SerializerT& s, std::queue<T> q) {
+  while(!q.empty()) {
+    s | q.front();
+    q.pop();
+  }
+}
+
+template <
+  typename SerializerT,
+  typename T,
+  typename = std::enable_if_t<
+    not std::is_same<
+      SerializerT,
+      checkpoint::Footprinter
+    >::value
```

Comment:
Use `std::is_same_v` (C++17):
```suggestion
    not std::is_same_v<
      SerializerT,
      checkpoint::Footprinter
    >
```

It may even fit in a single line now:
```suggestion
    not std::is_same_v<SerializerT, checkpoint::Footprinter>
```

Reply:
Good idea! I updated the code